h2o-algos/src/main/java/hex/glm/GLM.java

```diff
@@ -1023,6 +1028,15 @@ public void init(boolean expensive) {
               _parms.makeImputer(), 
               false, hasWeightCol(), hasOffsetCol(), hasFoldCol(), _parms.interactionSpec());
 
+      if (_parms._generate_variable_inflation_factors) {
+        _predictorNames = extractPredictorNames(_parms, _dinfo, _parms._fold_column);
+        // only calculate VIF for numerical columns
+        _validPredictors = Stream.of(_predictorNames).filter(x -> _parms.train().vec(x).isNumeric()).
```

Comment:
I think this looks like incorrect use of _parms.train() - because training frame gets adapted and some features dropped you can end-up with set of predictors that are not found in actual training frame - imho you need to use _train instead